### DISCUSSION TITLE
Login throttling by username and ip address, and configurable attempts and decays

### DISCUSSION THREAD

**Discussion by underdpt** [filamentphp/filament#8930] [Ideas]
Hello,

I'm trying to customize login throttling, and I have a few suggestions for a PR if there's interest on it:

- Throttle by default by username and ip address: currently throttling is done only by ip address. This can cause unexpected lockouts when you have a group of users behind a NAT network (same IP for all) and one hits the rate limiter. Also, throttling by username and ip address would match what Laravel's starter kits are doing.
- Allow customization of max attempts and decay seconds (via panel `->login()` with new method parameters)
- Allow block time to be different from decay (don't know how difficult this would be, but it would be good to be able to setup a block time of 1 hour if the rate limiter got 5 hits on 5 minutes, which now isn't possible).

This would help business to have Filament follow their security standards.

**Comment by playerro** [comment]
@underdpt 

> Allow customization of max attempts and decay seconds (via panel ->login() with new method parameters)

Will be great. Now you have to override huge authenticate() method to achieve that
